### PR TITLE
Update samples and tests to target netcoreapp2.1

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

**Comment by pranavkm** [comment]
Follow up : https://github.com/aspnet/AzureIntegration/issues/120

File: test/ApplicationInsights.HostingStartup.Tests/JavaScriptSnippetTest.cs [hunk 1]
```diff
@@ -17,15 +17,19 @@ public JavaScriptSnippetTest(ITestOutputHelper output) : base(output)
...
-        {
-            await JavaScriptSnippetInjectionTestSuite(applicationType);
-        }
+        [Fact]
+        public Task ScriptIsInjected_ForNetCoreApp20_Standalone() => JavaScriptSnippetInjectionTestSuite("netcoreapp2.0", ApplicationType.Standalone);
+
+        [Fact]
+        public Task ScriptIsInjected_ForNetCoreApp20_Portable() => JavaScriptSnippetInjectionTestSuite("netcoreapp2.0", ApplicationType.Portable);
+
+        [Fact(Skip = "Waiting for Kestrel")]
```
Comment (pranavkm) [inline test/ApplicationInsights.HostingStartup.Tests/JavaScriptSnippetTest.cs]: Undo if @pakrym gets Kestrel changes in.

File: build/dependencies.props [hunk 2]
```diff
@@ -5,29 +5,29 @@
...
+    <MicrosoftAspNetCoreDataProtectionAzureKeyVaultPackageVersion>2.1.0-preview1-27579</MicrosoftAspNetCoreDataProtectionAzureKeyVaultPackageVersion>
+    <MicrosoftAspNetCoreHostingAbstractionsPackageVersion>2.1.0-preview1-27579</MicrosoftAspNetCoreHostingAbstractionsPackageVersion>
+    <MicrosoftAspNetCoreHostingPackageVersion>2.1.0-preview1-27579</MicrosoftAspNetCoreHostingPackageVersion>
+    <MicrosoftAspNetCoreIdentityServiceAzureKeyVaultPackageVersion>2.1.0-preview1-27579</MicrosoftAspNetCoreIdentityServiceAzureKeyVaultPackageVersion>
+    <MicrosoftAspNetCoreMvcPackageVersion>2.1.0-preview1-27579</MicrosoftAspNetCoreMvcPackageVersion>
+    <MicrosoftAspNetCoreRazorRuntimePackageVersion>2.1.0-preview1-27579</MicrosoftAspNetCoreRazorRuntimePackageVersion>
+    <MicrosoftAspNetCoreServerIISIntegrationPackageVersion>2.1.0-preview1-27579</MicrosoftAspNetCoreServerIISIntegrationPackageVersion>
+    <MicrosoftAspNetCoreServerIntegrationTestingPackageVersion>0.5.0-preview1-27579</MicrosoftAspNetCoreServerIntegrationTestingPackageVersion>
+    <MicrosoftAspNetCoreServerKestrelPackageVersion>2.1.0-preview1-27579</MicrosoftAspNetCoreServerKestrelPackageVersion>
```
Comment (pranavkm) [inline build/dependencies.props]: @ryanbrandenburg \ @natemcmaster are there any consequences to manually updating this version? We'll need these for any netcoreapp2.1 tests running on kestrel.

Comment (natemcmaster) [reply build/dependencies.props]: nope, go for it

**Pull request merged**